keras_nlp/utils/text_generation.py

```diff
@@ -393,3 +393,145 @@ def token_probability_fn(inputs):
     if input_is_1d:
         return tf.squeeze(prompt)
     return prompt
+
+
+def top_p_search(
+    token_probability_fn,
+    prompt,
+    max_length,
+    p,
+    seed=None,
+    from_logits=False,
+    end_token_id=None,
+    pad_token_id=0,
+):
+    """
+    Text generation utility based on top-p (nucleus) sampling.
+
+    Top-p search filters the top token with probabilities that sum up to p, and
```

Comment:
maybe we can explain this a little more clearly, also remember to backtick argument names so it's clear what they are...

Top-p search selects tokens from the smallest subset of output probabilities that sum to greater than `p`. Put another way, top-p will first order token predictions by likelihood, and ignore all tokens after the cumulative probably of selected tokens exceeds `p`.